pandas/core/algorithms.py

```diff
@@ -706,6 +706,17 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
                 keys = Index(keys)
             result = Series(counts, index=keys, name=name)
 
+            # Use same index as the original values
+            if result.index.isna().sum() > 0:
+                fill_value = result[result.index.isna()].values[0]
+                result = result.reindex(unique(values), fill_value=fill_value)
```

Comment:
huh? this is way beyond scope

Reply:
The problem is that `_value_counts_arraylike` replaces all generic nan values (e.g. `None`) with `np.NaN` and reindexing has a mismatch between the original `None` and the new `np.NaN` in the index. The fill_value replaces the new missing value with the original value again.

Alternatively, it is also possible to use a generic `values = values.fillna(np.NaN)` to replace the `None`s with `np.NaN`s if there are nans in the index. I'll push a follow up commit about this in a bit. Would that be better than the above?